eks/main.tf

```diff
@@ -1,34 +1,38 @@
+locals {
+  all_domains = toset(compact(distinct(concat([var.base_domain], var.other_domains)))) 
+} 
+
 data "aws_route53_zone" "this" {
-  count = var.base_domain == null ? 0 : 1
+  for_each = local.all_domains
 
-  name = var.base_domain
+    name = each.key
```

Comment:
seems like there are lintint errors, can you run `terraform fmt -recursive` ?

Reply:
done